New exit condition

nmartinkova opened the pull request:
Stop measurement after the last gate is triggered. Do not wait for all gates to get triggered, as animals might jump over some gates and then return to them later.

nmartinkova committed:
New exit condition

Stop measurement after the last gate is triggered. Do not wait for all gates to get triggered.

tomas-fryza commented:
Dobré dopoledne,
domnívám se, že testování proměnné "i" v cyklu nebude mít žádný význam, respektive že tato podmínka bude vždy splněna.
Tom Frýza
Get BlueMail <https://bluemail.me> for Desktop
Natália Martínková wrote:
Stop measurement after the last gate is triggered. Do not wait for all gates to get triggered, as animals might jump over some gates and then return to them later.
You can view, comment on, or merge this pull request online at:
https://github.com/tomas-fryza/arduino-lizards/pull/2 <https://github.com/tomas-fryza/arduino-lizards/pull/2> 
Commit Summary
a12d4e1 <https://github.com/tomas-fryza/arduino-lizards/pull/2/commits/a12d4e13d04f35237b08e4d797caf8f513a4dd69> New exit condition
File Changes 
(1 file <https://github.com/tomas-fryza/arduino-lizards/pull/2/files> )
M software/lizards/src/main.cpp <https://github.com/tomas-fryza/arduino-lizards/pull/2/files#diff-3f04c7e88c127dd3f763a3e610e79b7aeb7dcd77eb5b34588dc950887f4907d1> (2) 
Patch Links:
https://github.com/tomas-fryza/arduino-lizards/pull/2.patch <https://github.com/tomas-fryza/arduino-lizards/pull/2.patch> 
https://github.com/tomas-fryza/arduino-lizards/pull/2.diff <https://github.com/tomas-fryza/arduino-lizards/pull/2.diff> 
—
Reply to this email directly, view it on GitHub <https://github.com/tomas-fryza/arduino-lizards/pull/2> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ALVSTHKRV33ZCQQUVEMD23LXGRUKFANCNFSM6AAAAAAYETCJGY> .
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

nmartinkova commented:
Máte pravdu. Chcela by som, aby meranie skončilo, keď jašterica prejde poslednou bránou (15 podľa riadku 28). Stáva sa zriedka, že by sme mali záznamy pre všetky brány a v chladnom počasí nám zvieratá za tú minútu čakania prechladnú. Neviem, ako to urobiť.

tomas-fryza commented:
V tom případě přidat ještě jednu podmínku, kdy smyčka má skončit:
* jsou přerušeny všechny měřící dráhy ... to by byl ideální případ
* uplyne čas, nyní nastaven na 60 sekund
* byla přerušena POSLEDNÍ brána č. 15
} while ((measured_num < NGATES) && (current_millis - start_millis <= TIMEOUT_MILLIS) && (measured_status[NGATES-1]));
Pokud by ještěrka "přeskočila" poslední bránu a už byla v cíli, stačilo by pak prstem onu poslední bránu přerušit a měření bby skončilo.
Tom Frýza
Get BlueMail <https://bluemail.me> for Desktop
Natália Martínková wrote:
Máte pravdu. Chcela by som, aby meranie skončilo, keď jašterica prejde poslednou bránou (15 podľa riadku 28). Stáva sa zriedka, že by sme mali záznamy pre všetky brány a v chladnom počasí nám zvieratá za tú minútu čakania prechladnú. Neviem, ako to urobiť.
—
Reply to this email directly, view it on GitHub <https://github.com/tomas-fryza/arduino-lizards/pull/2#issuecomment-1562576434> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ALVSTHJ2VDV7LYHD6F4T3QDXH4QMDANCNFSM6AAAAAAYETCJGY> .
You are receiving this because you commented.Message ID: ***@***.***>

nmartinkova committed:
Update exit condition

Edit copied from discussion to ensure end of measurement when the last gate is crossed
